api/all/src/main/java/io/opentelemetry/api/common/ArrayBackedAttributes.java

```diff
@@ -26,6 +26,16 @@ private ArrayBackedAttributes(Object[] data, Comparator<AttributeKey<?>> keyComp
     super(data, keyComparator);
   }
 
+  /**
+   * Only use this constructor if you can guarantee that the data has been de-duped, sorted by key
+   * and contains no null values or null/empty keys.
+   *
+   * @param data the raw data
+   */
+  ArrayBackedAttributes(Object[] data) {
+    super(data, KEY_COMPARATOR_FOR_CONSTRUCTION);
```

Comment:
Doesn't `super` still call `sortAndFilter` for our presorted ones? We could probably have the others explicitly pass in `Comparator.naturalOrder` and implement the one-arg super constructor as a simple field set

Reply:
oh drat. You are totally correct. Yes, I will refactor this a bit.

Reply:
fixed. please take a look.